modules/homa/src/lib.rs

```diff
@@ -16,118 +16,1065 @@
+	pub type StakingLedgers<T: Config> = StorageMap<_, Twox64Concat, u16, StakingLedger, OptionQuery>;
+
+	/// The total staking currency to bond on relaychain when new era,
+	/// and that is available to be match fast redeem request.
+	/// ToBondPool value: StakingCurrencyAmount
+	#[pallet::storage]
+	#[pallet::getter(fn to_bond_pool)]
+	pub type ToBondPool<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+	/// The total amount of void liquid currency. It's will not be issued,
+	/// used to avoid newly issued LDOT to obtain the incoming staking income from relaychain.
+	/// And it is guaranteed that the current exchange rate between liquid currency and staking
+	/// currency will not change. It will be reset to 0 at the beginning of the `rebalance` when new
+	/// era starts.
+	///
+	/// TotalVoidLiquid value: LiquidCurrencyAmount
+	#[pallet::storage]
+	#[pallet::getter(fn total_void_liquid)]
+	pub type TotalVoidLiquid<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+	/// The total unclaimed redemption.
+	///
+	/// UnclaimedRedemption value: StakingCurrencyAmount
+	#[pallet::storage]
+	#[pallet::getter(fn unclaimed_redemption)]
+	pub type UnclaimedRedemption<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+	/// Requests to redeem staked currencies.
+	///
+	/// RedeemRequests: Map: AccountId => Option<(liquid_amount: Balance, allow_fast_match: bool)>
+	#[pallet::storage]
+	#[pallet::getter(fn redeem_requests)]
+	pub type RedeemRequests<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, (Balance, bool), OptionQuery>;
+
+	/// The records of unbonding by AccountId.
+	///
+	/// Unbondings: double_map AccountId, ExpireEraIndex => UnbondingStakingCurrencyAmount
+	#[pallet::storage]
+	#[pallet::getter(fn unbondings)]
+	pub type Unbondings<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, T::AccountId, Twox64Concat, EraIndex, Balance, ValueQuery>;
+
+	/// The estimated staking reward rate per era on relaychain.
+	///
+	/// EstimatedRewardRatePerEra: value: Rate
+	#[pallet::storage]
+	#[pallet::getter(fn estimated_reward_rate_per_era)]
+	pub type EstimatedRewardRatePerEra<T: Config> = StorageValue<_, Rate, ValueQuery>;
+
+	/// Th maximum amount of bonded staking currency for a single sub on relaychain to obtain the
+	/// best staking rewards.
+	///
+	/// SoftBondedCapPerSubAccount: value: Balance
+	#[pallet::storage]
+	#[pallet::getter(fn soft_bonded_cap_per_sub_account)]
+	pub type SoftBondedCapPerSubAccount<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+	/// Th staking amount of threshold to mint.
+	///
+	/// MintThreshold: value: Balance
+	#[pallet::storage]
+	#[pallet::getter(fn mint_threshold)]
+	pub type MintThreshold<T: Config> = StorageValue<_, Balance, ValueQuery>;
+
+	/// Th liquid amount of threshold to redeem.
+	///
+	/// RedeemThreshold: value: Balance
+	#[pallet::storage]
+	#[pallet::getter(fn redeem_threshold)]
+	pub type RedeemThreshold<T: Config> = StorageValue<_, Balance, ValueQuery>;
```

Comment:
This is something we unlikely need to change, so could make it a constant instead.